pyomo/common/dependencies.py

```diff
@@ -17,22 +17,19 @@
 import warnings
 
 from .deprecation import deprecated, deprecation_warning, in_testing_environment
+from .errors import DeferredImportError
 from . import numeric_types
 
 
 SUPPRESS_DEPENDENCY_WARNINGS = False
 
 
-class DeferredImportError(ImportError):
-    pass
-
-
 class ModuleUnavailable(object):
-    """Mock object that raises a DeferredImportError upon attribute access
+    """Mock object that raises :py:class:`.DeferredImportError` upon attribute access
```

Comment:
Maybe this is nothing, but you switch between `.DeferredImportError` and `DeferredImportError` (no period) several times in this documentation. Is that intentional?

Reply:
The "`.`" causes Sphinx to correctly link to the documentation.  We should probably change it "everywhere" .. but before we go down that rabbit hole, we should probably all get together and decide on what the documentation layout *should* look like, and what convention we want to use for cross-linking.

Reply:
_A design discussion, you say?_ I'm all for it. If only we had time.